### PR TITLE
Added MD4 digest and testcase

### PR DESCRIPTION
**Pull request by damon-kwok**
Added MD4 digest and testcase

### CODE REVIEW

**Comment by SeanTAllen** [comment]
@mfelsche you merged this without a CHANGELOG label, please manually add a CHANGELOG entry.

**Comment by SeanTAllen** [comment]
Why was the MD5 test case removed?

**Comment by SeanTAllen** [comment]
MD4 has security problems. I think this should be removed. I don't see any reason to support it.

File: crypto/_test.pony [hunk 1]
```diff
@@ -106,19 +106,26 @@ class iso _TestDigest is UnitTest
...
       "94af09c09bb9bb7b5c94fec6e6121482",
-      ToHexString(d.final()))
+      ToHexString(d'.final()))
+    h.assert_eq[String](
+      "94af09c09bb9bb7b5c94fec6e6121482",
+      ToHexString(d'.final()))
 
-    let d' = Digest.sha512()
-    d'.append("test")?
+    let d'' = Digest.sha512()
```
Comment (SeanTAllen) [inline crypto/_test.pony:126]: this is a pretty awful name. `d''`, can you clean this up and come up with some better names for these?

distinguishing

d, d', and d'' is not easy. 

doing d'' is abusing being allowed to add prime to variables in my opinion.

Comment (damon-kwok) [reply crypto/_test.pony:126]: Don't worry, they're still here. They're just changing names
https://github.com/ponylang/crypto/blob/1479d97c16de6480f68a5b5993110cb79899ceaa/crypto/_test.pony#L123

Comment (damon-kwok) [reply crypto/_test.pony:126]: NP, I will maintain it for a long time, they will get better soon

Comment (SeanTAllen) [reply crypto/_test.pony:126]: We can't assume when we merge things who will be maintaining them.

Comment (damon-kwok) [reply crypto/_test.pony:126]: You are correct.

**Pull request merged**